apps/src/applab/designMode.js

```diff
@@ -469,7 +469,38 @@ function makeDraggable (jqueryElements) {
         // resizable sets z-index to 90, which we don't want
         $(this).children().css('z-index', '');
       },
-      resize: function () {
+      resize: function (event, ui) {
+        // Wishing for a vector maths library...
+
+        // Customize motion according to current visualization scale.
+        var div = document.getElementById('divApplab');
+        var xScale = div.getBoundingClientRect().width / div.offsetWidth;
+        var yScale = div.getBoundingClientRect().height / div.offsetHeight;
+        var deltaWidth = ui.size.width - ui.originalSize.width;
+        var deltaHeight = ui.size.height - ui.originalSize.height;
+        var newWidth = ui.originalSize.width + (deltaWidth / xScale);
+        var newHeight = ui.originalSize.height + (deltaHeight / yScale);
```

Comment:
Worth de-duping against https://github.com/code-dot-org/code-dot-org/pull/4311/files#diff-7f9b30159330297b3b2915dcafb93087L495 down in `.draggable`?